src/tokenpay/index.ts

```diff
@@ -0,0 +1,209 @@
+import { ClientOptions, InscriptionEnv, InscriptionEnvNetwork } from "../types";
+import { TokenPayClient } from "./client";
+import {
+  BitcoinNetworkType,
+  SignTransactionPayload,
+  SignTransactionResponse,
+  signTransaction,
+} from "sats-connect";
+import {
+  AccountBalanceResponse,
+  AccountWithdrawRequest,
+  AccountWithdrawResponse,
+  CheckTransactionAsTxidRequest,
+  CheckTransactionAsTxidResponse,
+  CreatePaymentPSBTRequest,
+  CreatePaymentPSBTResponse,
+  CreateRuneOrderRequest,
+  GetAccountWithdrawRequest,
+  GetOrderRequest,
+  GetOrderResponse,
+  RuneOrderResponse,
+  SatsConnectWrapperResponse,
+} from "../types/tokenpay_types";
+import { WALLET_PROVIDER } from "../types/marketplace_types";
+
+/**
+ * A class for interacting with the TokenPay API's.
+ */
+export class TokenPay {
+  /**
+   * The Bitcoin network type to use ('Mainnet', 'Testnet' or 'Signet').
+   */
+  private network: BitcoinNetworkType;
+  /**
+   * The instance of TokenPayClient used to make API requests.
+   */
+  private tokenpayClientInstance!: TokenPayClient;
+
+  /**
+   * Creates an instance of TokenPay.
+   * @param key The API key (optional).
```

Comment:
need to add both apikeys where tokenpay apikey is optional.
but tokenpay api key is not optional if user wants to init tokenpay.

Reply:
yes, it should throw if user tries to init tokenpay but tokenpay api key is not specified @T1-O

Reply:
@dannybabbev @ordinariusprof this is done